h/services/flag.py

```diff
@@ -114,6 +128,10 @@ def flag_counts(self, annotation_ids):
         flag_counts = {f.annotation_id: f.flag_count for f in query}
         missing_ids = set(annotation_ids) - set(flag_counts.keys())
         flag_counts.update({id_: 0 for id_ in missing_ids})
+
```

Comment:
I don't know that much about H to suggest an alternative (and it's out of the scope for this PR anyway) but there's probably a place a few layers up where we know if we are going to need flag counts etc where we could build an annotations query that could be lazy evaluated by the first layer that needs it.

Maybe not the case here but adding querying in deeper layers leads to querying (almost) the same thing multiple times and those could add up.

Reply:
Yeah we can know we don't need this if the person isn't a moderator. I could look at doing that.

Reply:
Actually looking at this it isn't trivial, as a user can be a moderator or not, per annotation. However this does suggest to me that we are probably issuing lots of group queries to work that out...